Sources/protoc-gen-grpc-swift/Generator-Client+AsyncAwait.swift

```diff
@@ -130,6 +130,115 @@ extension Generator {
+    self.withIndentation("extension \(self.asyncClientProtocolName)", braces: .curly) {
+      // 'Safe' calls.
+      for (i, method) in self.service.methods.enumerated() {
+        if i > 0 {
+          self.println()
+        }
+        self.method = method
+
+        let rpcType = streamingType(self.method)
+        let callTypeWithoutPrefix = Types.call(for: rpcType, withGRPCPrefix: false)
+        let responseStreamType = "GRPCAsyncResponseStream"
+
+        switch rpcType {
+        case .unary:
+          self.printFunction(
+            name: self.methodFunctionName,
+            arguments: [
+              "_ request: \(self.methodInputName)",
+              "callOptions: \(Types.clientCallOptions)? = nil",
+            ],
+            returnType: self.methodOutputName,
+            access: self.access,
+            async: true,
+            throws: true
+          ) {
+            self.withIndentation(
+              "return try await self.perform\(callTypeWithoutPrefix)",
+              braces: .round
+            ) {
+              self.println("path: \(self.methodPath),")
+              self.println("request: request,")
+              self.println("callOptions: callOptions ?? self.defaultCallOptions")
+            }
+          }
+
+        case .serverStreaming:
+          self.printFunction(
+            name: self.methodFunctionName,
+            arguments: [
+              "_ request: \(self.methodInputName)",
+              "callOptions: \(Types.clientCallOptions)? = nil",
+            ],
+            returnType: "\(responseStreamType)<\(self.methodOutputName)>",
+            access: self.access
+          ) {
+            self.withIndentation(
+              "return self.perform\(callTypeWithoutPrefix)",
+              braces: .round
+            ) {
+              self.println("path: \(self.methodPath),")
+              self.println("request: request,")
+              self.println("callOptions: callOptions ?? self.defaultCallOptions")
+            }
+          }
+
+        case .clientStreaming:
+          self.printFunction(
+            name: "\(self.methodFunctionName)<RequestStream>",
+            arguments: [
+              "requests: RequestStream",
+              "callOptions: \(Types.clientCallOptions)? = nil",
+            ],
+            returnType: self.methodOutputName,
+            access: self.access,
+            async: true,
+            throws: true,
+            genericWhereClause: "where RequestStream: AsyncSequence, RequestStream.Element == \(self.methodInputName)"
+          ) {
+            self.withIndentation(
+              "return try await self.perform\(callTypeWithoutPrefix)",
+              braces: .round
+            ) {
+              self.println("path: \(self.methodPath),")
+              self.println("requests: requests,")
+              self.println("callOptions: callOptions ?? self.defaultCallOptions")
+            }
+          }
+
+        case .bidirectionalStreaming:
```

Comment:
This is the same as `.clientStreaming` I think. We also need the `Sequence` variant codegen.

Reply:
See comment on other thread https://github.com/grpc/grpc-swift/pull/1261/files#r709267054.